PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = 'hr.contract'
-    _inherit = 'hr.employee'
-    _inherit = 'hr.work.entry'
-    _inherit = 'hr.work.entry.type'
-    _inherit = 'resource.calendar'

Total Changes: 11

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_work_entry_contract/models/hr_contract.py
+++ b/addons/hr_work_entry_contract/models/hr_contract.py
@@ -18,7 +18,7 @@ from .hr_work_intervals import WorkIntervals
 
 
 class HrContract(models.Model):
-    _inherit = 'hr.contract'
+    _inherit = ['hr.contract']
     _description = 'Employee Contract'
 
     date_generated_from = fields.Datetime(string='Generated From', readonly=True, required=True,

--- a/addons/hr_work_entry_contract/models/hr_employee.py
+++ b/addons/hr_work_entry_contract/models/hr_employee.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class HrEmployee(models.Model):
-    _inherit = 'hr.employee'
+    _inherit = ['hr.employee']
 
     def generate_work_entries(self, date_start, date_stop, force=False):
         date_start = fields.Date.to_date(date_start)

--- a/addons/hr_work_entry_contract/models/hr_work_entry.py
+++ b/addons/hr_work_entry_contract/models/hr_work_entry.py
@@ -12,7 +12,7 @@ from odoo.addons.hr_work_entry_contract.models.hr_work_intervals import WorkInte
 
 
 class HrWorkEntry(models.Model):
-    _inherit = 'hr.work.entry'
+    _inherit = ['hr.work.entry']
 
     contract_id = fields.Many2one('hr.contract', string="Contract", required=True)
     employee_id = fields.Many2one(domain=[('contract_ids.state', 'in', ('open', 'pending'))])
@@ -186,7 +186,7 @@ class HrWorkEntry(models.Model):
 
 
 class HrWorkEntryType(models.Model):
-    _inherit = 'hr.work.entry.type'
+    _inherit = ['hr.work.entry.type']
     _description = 'HR Work Entry Type'
 
     is_leave = fields.Boolean(

--- a/addons/hr_work_entry_contract/models/resource_calendar.py
+++ b/addons/hr_work_entry_contract/models/resource_calendar.py
@@ -2,8 +2,9 @@
 
 from odoo import models
 
+
 class ResourceCalendar(models.Model):
-    _inherit = 'resource.calendar'
+    _inherit = ['resource.calendar']
 
     def _get_global_attendances(self):
         return super()._get_global_attendances().filtered(lambda a: not a.work_entry_type_id.is_leave)
